### DISCUSSION TITLE
Be able to set paginated() per panel instead of per table/globally

### DISCUSSION THREAD

**Discussion by miguilimzero** [filamentphp/filament#7625] [Ideas]
I have an application with tons of resources and relation managers. This application is for the end user and not an admin panel. Some resources have tons of data, and I can't allow the user to show all rows at once. **I would not like to set it globally, as there is an administration panel in the same project.**

It would be good to set the `paginated()` function per panel instead of [per table](https://filamentphp.com/docs/3.x/tables/advanced#customizing-the-pagination-options) or globally. Something like this:

```php
public function panel(Panel $panel): Panel
{
        return $panel
            ->paginated([10, 25, 50, 100]);
            // All other panel settings...
}
```

**Comment by danharrin** [comment]
```php
public function panel(Panel $panel): Panel
{
        return $panel
            ->bootUsing(function () {
                Table::configureUsing(fn ($table) => $table->[10, 25, 50, 100]);
            });
}
```

**Reply by leandrocfe** [reply]
`->bootUsing()` is insane!  🤯  🦒

**Reply by wickham-c** [reply]
I couldn't get the above code to run in v3 due to a `syntax error, unexpected token "["`, so I went digging in the [docs](https://filamentphp.com/docs/3.x/tables/advanced#global-settings) and found that this works...
```php
public function panel(Panel $panel): Panel
{
    return $panel
        ->bootUsing(function () {
            Table::configureUsing(fn ($table) => $table->paginationPageOptions([10, 25, 50, 100]));
        });
}
```

**Reply by smiliyas** [reply]
Try this one:

```PHP
public function panel(Panel $panel): Panel
{
    return $panel
        ->bootUsing(function () {
            Table::configureUsing(fn ($table) => $table->paginated([10, 25, 50, 100]));
        });
}
```

**Reply by wickham-c** [reply]
The `paginated()` method defined in vendor/filament/tables/src/Table/Concerns/**CanPaginateRecords.php** calls the `paginationPageOptions()` method, passing its argument in, so either method ultimately achieves the same goal.

**Reply by vishnupeas** [reply]
This should be added in docs, filament table's pagination section.
